### ISSUE TITLE
irepl --rlock fails with 'file name too long' when irepl by itself doesn't

### ISSUE THREAD

**Issue by tedgin** [irods/irods#4433]
- [x] master
- [x] 4-2-stable
--------

### iRODS Version, OS and Version
I'm using version 4.1.10 of irepl with the same version server.  I'm also using CentOS 7 on both client and server.

### What did you try to do?

```
irepl -B -M -v --rlock -R taccCorralRes \ 
/iplant/home/1234567890/Nine_Hosts_fastq/Mapped_to_CloneO_remove_duplicates/RMTA_v2.5_Nine_Hosts-2019-04-16-08-30-34.5/Nine-hosts/rmdup.bam/Samtools_1.7_Merge_Maize_composed_nonSort-2019-06-03-16-17-34.6/GATK_HaplotypeCaller_3.8.0_Maize-2019-06-08-14-26-14.0/gatk_haplotypecaller_output/Maize.sorted.rmdup.vcf
```

The username `1234567890` is not the actual username, but it has the same number of characters.

### Expected behavior
I expected a replica to be added to the `taccCorralRes` resource.

### Observed behavior (including steps to reproduce, if applicable)
I got the following error in the terminal.

```
ERROR: replUtil: repl error for /iplant/home/1234567890/Nine_Hosts_fastq/Mapped_to_CloneO_remove_duplicates/RMTA_v2.5_Nine_Hosts-2019-04-16-08-30-34.5/Nine-hosts/rmdup.bam/Samtools_1.7_Merge_Maize_composed_nonSort-2019-06-03-16-17-34.6/GATK_HaplotypeCaller_3.8.0_Maize-2019-06-08-14-26-14.0/gatk_haplotypecaller_output/Maize.sorted.rmdup.vcf, status = -900036 status = -900036 FILE_OPEN_ERR, File name too long
```

In the IES log, I got the following messages.

```
Jul  1 08:46:10 pid:69659 ERROR: fsDataObjLock: open error for /iplant/home/1234567890/Nine_Hosts_fastq/Mapped_to_CloneO_remove_duplicates/RMTA_v2.5_Nine_Hosts-2019-04-16-08-30-34.5/Nine-hosts/rmdup.bam/Samtools_1.7_Merge_Maize_composed_nonSort-2019-06-03-16-17-34.6/GATK_HaplotypeCaller_3.8.0_Maize-2019-06-08-14-26-14.0/gatk_haplotypecaller_output/Maize.sorted.rmdup.vcf status = -900036 FILE_OPEN_ERR, File name too long
Jul  1 08:46:10 pid:69659 ERROR: rsDataObjRepl: lock error for /iplant/home/1234567890/Nine_Hosts_fastq/Mapped_to_CloneO_remove_duplicates/RMTA_v2.5_Nine_Hosts-2019-04-16-08-30-34.5/Nine-hosts/rmdup.bam/Samtools_1.7_Merge_Maize_composed_nonSort-2019-06-03-16-17-34.6/GATK_HaplotypeCaller_3.8.0_Maize-2019-06-08-14-26-14.0/gatk_haplotypecaller_output/Maize.sorted.rmdup.vcf. lockType = readLockType status = -900036 FILE_OPEN_ERR, File name too long
Jul  1 08:46:10 pid:69659 NOTICE: readAndProcClientMsg: received disconnect msg from client
Jul  1 08:46:10 pid:69659 NOTICE: Agent exiting with status = 0
```

After seeing the error was related to file locking, I removed `--rlock` from the command, and it succeeded.

```
irepl -B -M -v -R taccCorralRes /iplant/home/1234567890/Nine_Hosts_fastq/Mapped_to_CloneO_remove_duplicates/RMTA_v2.5_Nine_Hosts-2019-04-16-08-30-34.5/Nine-hosts/rmdup.bam/Samtools_1.7_Merge_Maize_composed_nonSort-2019-06-03-16-17-34.6/GATK_HaplotypeCaller_3.8.0_Maize-2019-06-08-14-26-14.0/gatk_haplotypecaller_output/Maize.sorted.rmdup.vcf
   Maize.sorted.rmdup.vcf       1038.548 MB | 10.851 sec | 16 thr | 95.707 MB/s
```

**Comment by alanking** [comment]
The same thing happens when using `--wlock` with `iput`:
```
$ iput VERSION.json dir_000000/dir_000001/dir_000002/dir_000003/dir_000004/dir_000005/dir_000006/dir_000007/dir_000008/dir_000009/dir_000010/dir_000011/dir_000011/dir_000012/dir_000013/dir_000014/dir_000015/dir_000016/dir_000017/dir_000018/dir_000019/dir_000020/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
$ ils -l dir_000000/dir_000001/dir_000002/dir_000003/dir_000004/dir_000005/dir_000006/dir_000007/dir_000008/dir_000009/dir_000010/dir_000011/dir_000011/dir_000012/dir_000013/dir_000014/dir_000015/dir_000016/dir_000017/dir_000018/dir_000019/dir_000020/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
  rods              0 demoResc       372442 2019-10-10.19:09 & aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
$ iput --wlock -f file.txt dir_000000/dir_000001/dir_000002/dir_000003/dir_000004/dir_000005/dir_000006/dir_000007/dir_000008/dir_000009/dir_000010/dir_000011/dir_000011/dir_000012/dir_000013/dir_000014/dir_000015/dir_000016/dir_000017/dir_000018/dir_000019/dir_000020/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
remote addresses: 172.17.0.2 ERROR: putUtil: put error for /tempZone/home/rods/dir_000000/dir_000001/dir_000002/dir_000003/dir_000004/dir_000005/dir_000006/dir_000007/dir_000008/dir_000009/dir_000010/dir_000011/dir_000011/dir_000012/dir_000013/dir_000014/dir_000015/dir_000016/dir_000017/dir_000018/dir_000019/dir_000020/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa, status = -900036 status = -900036 FILE_OPEN_ERR, File name too long
```
In the same way, the error is being caused by the creation of the "lock file":
https://github.com/irods/irods/blob/ed64810bcfffe8ab9b02b85747b40884fbe27a53/server/core/src/physPath.cpp#L1291-L1298

**Comment by kript** [comment]
We are seeing this too with `iput`; one my my bioinformatics colleague will hopefully be along shortly with a reproducer, but any chance of getting a fix in 4.2.7?

**Comment by FMakosza** [comment]
Can confirm it's an issue with `iput --wlock` as well. Looks like the problem occurs when the full path of the file is longer than 260 characters exactly?

First, for `/humgen/teams/hgi/`:
```
$ iput --wlock 123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_12345
ERROR: putUtil: put error for /humgen/teams/hgi/123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_12345, status = -900036 status = -900036 FILE_OPEN_ERR, File name too long
$ iput 123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_12345
$ ils
/humgen/teams/hgi:
  123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_12345

$ iput --wlock 123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_1
$ ils
/humgen/teams/hgi:
  123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_1

```
Then, for `/humgen/teams/hgi/test/`:
```
$ iput --wlock 123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_1                   
ERROR: putUtil: put error for /humgen/teams/hgi/test/123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_1, status = -9
00036 status = -900036 FILE_OPEN_ERR, File name too long
```
The file that `iput --wlock`ed successfully into `/humgen/teams/hgi/` causes an error. Five characters need to be trimmed off the end to compensate for `/test` before it can be `iput --wlock`ed successfully again.
```
$ iput --wlock 123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456
$ ils
/humgen/teams/hgi/test:
  123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456789_123456
```

**Comment by alanking** [comment]
This will be resolved in #5486 as the file locks will no longer be used in the server.

**Comment by alanking** [comment]
Cannot check boxes, but I think this can be closed.
